tensorboard/writer/event_file_writer.py

```diff
@@ -0,0 +1,229 @@
+# Copyright 2015 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Writes events to disk in a logdir."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os.path
+import socket
+import threading
+import time
+
+import six
+
+from tensorboard.compat.proto import event_pb2
+from tensorboard.writer.record_writer import RecordWriter, directory_check
+
+
+class EventsWriter(object):
```

Comment:
I agree. BTW, we have some issues regarding the python threading library (which seldom happens). It looks like the only solution is changing the underlying writing procedure from `threading` to `multiprocess`. Considering the trade-offs, do you recommend switching to multiprecessing? ref. https://github.com/lanpa/tensorboardX/issues/123

Reply:
Hmm, it sounds like this issue happens only when using a single `Writer` from multiple processes at once, which this doesn't work because `fork()` is broken for multi-threaded code (see e.g. https://thorstenball.com/blog/2014/10/13/why-threads-cant-fork/).

I don't think we really want to support that use case.  If somebody is using multiprocessing to parallelize training, they should use a separate Writer instance in each process.  Otherwise they will have far more problems than just deadlocks as described in the issue - for example, the same Writer across multiple forked processes is presumably using the same file descriptor, but writes to it aren't synchronized (because locks don't have any effect for multiprocess code) so it would be possible to end up with conflicting or overlapping writes to the same file, resulting in data corruption.